src/ThriftSQL/ImpalaQuery.php

```diff
@@ -51,35 +53,47 @@ public function wait() {
 
     } while (true);
 
+    // Wait for results by fetching some rows -- triggers query to run
+    $this->_fetchResponse( 2 );
```

Comment:
I'm not sure how to test it tbh, but wouldn't this "consume" those 2 results as well as waiting? Such that when you `fetch` afterwards, you won't get those 2 initial results used for waiting?

Reply:
So that's why `_fetchResponse()` stores the result in `$this->_lastResponse` and when `fetch()` gets called it consumes that response first.